controllers/binding/snapshotenvironmentbinding_controller.go

```diff
@@ -82,10 +83,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
-	application, err := r.getApplicationFromSnapshotEnvironmentBinding(ctx, snapshotEnvironmentBinding)
+	var application *applicationapiv1alpha1.Application
+	err = retry.OnError(retry.DefaultRetry, func(_ error) bool { return true }, func() error {
+		application, err = r.getApplicationFromSnapshotEnvironmentBinding(ctx, snapshotEnvironmentBinding)
+		return err
+	})
 	if err != nil {
-		logger.Error(err, "Failed to get Application from the SnapshotEnvironmentBinding")
-		return ctrl.Result{}, err
+		return logger.HandleLoaderError(err, "Application", "SnapshotEnvironmentBinding")
```

Comment:
Logger shouldn't have responsibility to decide code flow (yes there is exception in go, for `panic` and `fatal` but that ends everything)

https://github.com/redhat-appstudio/integration-service/pull/329

Reply:
#342